Web/src/components/RoutesWrapper.jsx

```diff
@@ -0,0 +1,71 @@
+// src/RoutesWrapper.jsx
+import React, { useEffect, useState } from 'react';
+import { Route, Routes, useNavigate } from 'react-router-dom';
+import supabase from '../supabaseClient';
+import Dashboard from '../pages/Dashboard';
+import Expenses from '../pages/Expenses';
+import Budget from '../pages/Budget';
+import Goals from '../pages/Goals';
+import Authentication from '../pages/Authentication';
+import SignUp from '../pages/SignUp';
+import ForgotPassword from '../pages/ForgotPassword';
+import ResetPassword from '../pages/ResetPassword';
+
+
+function RoutesWrapper() {
+  const [loading, setLoading] = useState(true);
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+
+      // Check if the user is logged in when the app loads
+    const checkUserSession = async () => {
+      const { data: { session } } = await supabase.auth.getSession();
+
+      if (session) {
+        setIsAuthenticated(true);
+        // Only redirect to the dashboard if NOT on the password reset page
+        if (location.pathname !== '/reset-password' && location.pathname !== '/login' && location.pathname !== '/signup' && location.pathname !== '/forgot-password') {
```

Comment:
if you really wanted to you can move these into a different const and place them into here for more code modularity as this is used twice in this file